src/coreclr/vm/classlayoutinfo.cpp

```diff
@@ -226,23 +226,15 @@ namespace
         *pLargestAlignmentRequirementOut = LargestAlignmentRequirement;
     }
 
-    //=======================================================================
-    // This function returns TRUE if the provided corElemType disqualifies
-    // the structure from being a managed-sequential structure.
-    // The fsig parameter is used when the corElemType doesn't contain enough information
-    // to successfully determine if this field disqualifies the type from being
-    // managed-sequential.
-    // This function also fills in the pManagedPlacementInfo structure for this field.
-    //=======================================================================
-    BOOL CheckIfDisqualifiedFromManagedSequential(CorElementType corElemType, MetaSig& fsig, RawFieldPlacementInfo* pManagedPlacementInfo)
+    RawFieldPlacementInfo GetFieldPlacementInfo(CorElementType corElemType, TypeHandle pNestedType)
     {
-        pManagedPlacementInfo->m_alignment = TARGET_POINTER_SIZE;
-        pManagedPlacementInfo->m_size = TARGET_POINTER_SIZE;
+        // Initialize offset to a dummy value as we set it to the correct value later.
+        RawFieldPlacementInfo placementInfo { (UINT32)-1, TARGET_POINTER_SIZE, TARGET_POINTER_SIZE };
```

Comment:
I think the existing code was easier to read. With this change, one has to go a lookup the definition of `RawFieldPlacementInfo` to figure out what is getting initialized.